virttest/tests/unattended_install.py

```diff
@@ -1159,34 +1159,58 @@ def string_in_serial_log(serial_log_file_path, string):
+    sessions = []
+    close = []
+    try:
         try:
-            paths = console.cmd("ls -1 %s" % path_glob)
+            session = vm.wait_for_login()
+            close.append(session)
+            sessions.append(session)
         except Exception as details:
-            continue
-        for path in paths.splitlines():
-            if not path:
+            pass
+        if vm.serial_console:
+            sessions.append(vm.serial_console)
+        for i, console in enumerate(sessions):
+            try:
+                console.cmd("true")
+            except Exception as details:
+                logging.info("Skipping log_useful_files #%s: %s", i, details)
                 continue
-            if path.startswith(os.path.sep):
-                rel_path = path[1:]
-            else:
-                rel_path = path
-            dst = os.path.join(test.outputdir, vm.name, rel_path)
-            dst_dir = os.path.dirname(dst)
-            if not os.path.exists(dst_dir):
-                os.makedirs(dst_dir)
-            with open(dst, 'wb') as fd_dst:
+            for path_glob in ["/*.log", "/tmp/*.log", "/var/tmp/*.log"]:
                 try:
-                    fd_dst.write(console.cmd("cat %s" % path))
-                    logging.info("Attached '%s' log file from guest", dst_dir)
+                    paths = console.cmd("ls -1 %s" % path_glob)
                 except Exception as details:
-                    logging.warning("Unknown exception while attempt_to_log_"
-                                    "useful_files(): %s", details)
-                    fd_dst.write("Unknown exception while getting content: %s"
-                                 % details)
+                    continue
+                for path in paths.splitlines():
+                    if not path:
+                        continue
+                    if path.startswith(os.path.sep):
+                        rel_path = path[1:]
+                    else:
+                        rel_path = path
+                    dst = os.path.join(test.outputdir, vm.name, str(i),
+                                       rel_path)
+                    dst_dir = os.path.dirname(dst)
+                    if not os.path.exists(dst_dir):
+                        os.makedirs(dst_dir)
+                    with open(dst, 'w') as fd_dst:
+                        try:
+                            fd_dst.write(console.cmd("cat %s" % path))
+                            logging.info("Attached '%s' log file from guest",
+                                         dst_dir)
+                        except Exception as details:
+                            logging.warning("Unknown exception while "
+                                            "attempt_to_log_useful_files(): "
+                                            "%s", details)
+                            fd_dst.write("Unknown exception while getting "
+                                         "content: %s" % details)
+    finally:
```

Comment:
It look like if we have more than on valid session (there's no break to the `for i, console in enumerate(sessions):` line) we can end up with multiple copies of the same file... is that so?

Reply:
yes, they'll be in different dst dir (I use the `str(i)` as part of the output path. This might be useful in case of unreliable connection, or temporary connection issues where one session might fail to get some files.

Reply:
... but I'm not strictly bound to this. If you say you want to break after first successful execution, feel free to say so...

Reply:
Yes, if you're iterating over the files, then I think it's reasonable to break after the file is fetched from the first available "connection".

Reply:
OK